aptos-move/aptos-native-interface/src/context.rs

```diff
@@ -25,39 +28,39 @@ use std::{
 /// Major features include incremental gas charging and less ambiguous error handling. For this
 /// reason, native functions should always use [`SafeNativeContext`] instead of [`NativeContext`].
 #[allow(unused)]
-pub struct SafeNativeContext<'a, 'b, 'c> {
-    pub(crate) inner: &'c mut NativeContext<'a, 'b>,
+pub struct SafeNativeContext<'a, 'b, 'c, 'd> {
+    pub(crate) inner: &'d mut NativeContext<'a, 'b, 'c>,
 
-    pub(crate) timed_features: &'c TimedFeatures,
-    pub(crate) features: &'c Features,
+    pub(crate) timed_features: &'d TimedFeatures,
+    pub(crate) features: &'d Features,
     pub(crate) gas_feature_version: u64,
 
-    pub(crate) native_gas_params: &'c NativeGasParameters,
-    pub(crate) misc_gas_params: &'c MiscGasParameters,
-
-    pub(crate) gas_budget: InternalGas,
-    pub(crate) gas_used: InternalGas,
+    pub(crate) native_gas_params: &'d NativeGasParameters,
+    pub(crate) misc_gas_params: &'d MiscGasParameters,
 
-    pub(crate) enable_incremental_gas_charging: bool,
+    // Two fields below were used when there was no access to gas meter in native context. This is
+    // no longer the case, so these can be removed when the feature is stable.
+    pub(crate) legacy_gas_used: InternalGas,
+    pub(crate) legacy_enable_incremental_gas_charging: bool,
```

Comment:
Hm, can't this be an Option of internal gas? (it's used only if enable = true as I can see?)

Reply:
we have cases where it is false and we still accumulate gas? For table natives and Move stdlib we call 

```rust
pub fn with_incremental_gas_charging<F, R>(&mut self, enable: bool, action: F) -> R
    where
        F: FnOnce(&mut Self) -> R,
{
        let old = self.enable_incremental_gas_charging;
        self.enable_incremental_gas_charging = enable;
        let res = action(self);
        self.enable_incremental_gas_charging = old;
        res
}
```

Reply:
i.e., the gas is still charged, it just changes the granularity of the charges (for the whole function, or per `charge` call)